### PR TITLE
Rn 160 aggregation options endpoint

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
RN-160

### Changes:
1. Hard coded aggregation types' schema and description, named as `AGGREGATION_TYPES_META_DATA`, exposed from `@tupaia/aggregator` . 
2. Added an endpoint in `report-server` to expose `AGGREGATION_TYPES_META_DATA`.
3. Added an endpoint in `admin-panel-server` to request `AGGREGATION_TYPES_META_DATA` from `report-server` (support to fetch all and search by text). 
4. `jsonEditor` storybook.

As discussed with @IgorNadj , we agree this data flow is complicated, for now we try to avoid importing @tupaia/aggregator in `@tuapia/admin-panel-server`. Later it will be solved by moving AggregationTypes from `@tuapia/Aggregator` to `@tupaia/database` or even to a new package `@tupaia/types or @tupaia/models` (only contains types), so that we are safe to import in `@tuapia/admin-panel`.

### CODE REVIEW

File: packages/ui-components/src/components/JsonEditor/JsonEditor.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import JSONEditor from 'jsoneditor/dist/jsoneditor-minimalist';
+import JSONEditor from 'jsoneditor/dist/jsoneditor';
```
Comment (billli0) [inline packages/ui-components/src/components/JsonEditor/JsonEditor.js:3]: `jsoneditor-minimalist` do not support schema validation.

File: packages/aggregator/src/aggregationTypes.js [hunk 2]
```diff
@@ -32,41 +32,203 @@ const SUM_PREVIOUS_EACH_DAY = 'SUM_PREVIOUS_EACH_DAY';
...
+  },
+  {
+    code: MOST_RECENT,
+    description: 'Get the latest data within selected period',
+    schema: getSchema(MOST_RECENT, DATA_SOURCE_ENTITY_TYPE_REQUIRED_SCHEMA),
+  },
+  {
+    code: OFFSET_PERIOD,
+    description: 'Modify the period from the data',
+    schema: getSchema(OFFSET_PERIOD, OFFSET_PERIOD, OFFSET_PERIOD_SCHEMA),
```
Comment (IgorNadj) [inline packages/aggregator/src/aggregationTypes.js:166]: Does getSchema take 3 arguments or just 2?

**Review by IgorNadj — APPROVED** [review]
Looks good, pre-approving. Minor comments

File: packages/aggregator/src/aggregationTypes.js [hunk 3]
```diff
@@ -3,33 +3,72 @@
...
+const SUM_EACH_WEEK = 'SUM_EACH_WEEK';
+const SUM_MOST_RECENT_PER_FACILITY = 'SUM_MOST_RECENT_PER_FACILITY';
+const SUM_PER_ORG_GROUP = 'SUM_PER_ORG_GROUP';
+const SUM_PER_PERIOD_PER_ORG_GROUP = 'SUM_PER_PERIOD_PER_ORG_GROUP';
+const SUM_PREVIOUS_EACH_DAY = 'SUM_PREVIOUS_EACH_DAY';
+const SUM_UNTIL_CURRENT_DAY = 'SUM_UNTIL_CURRENT_DAY';
+const SUM = 'SUM';
+
+export const AGGREGATION_TYPES_META_DATA = [
+  { code: COUNT_PER_ORG_GROUP, description: 'COUNT_PER_ORG_GROUP' },
```
[the remark this answers is not in the document]
Comment (IgorNadj) [reply packages/aggregator/src/aggregationTypes.js]: See next commit

File: packages/aggregator/src/aggregationTypes.js (hunk 2 above)
Comment (billli0) [reply packages/aggregator/src/aggregationTypes.js]: Just 2, thanks for pointing.

**Comment by billli0** [comment]
Thanks for reviewing @IgorNadj . Latest commit has done a bit refactoring in `aggregationTypes.js` and removed the search functionality in the endpoint.

**Comment by IgorNadj** [comment]
> Thanks for reviewing @IgorNadj 🙏 . Latest commit has done a bit refactoring in `aggregationTypes.js` and removed the search functionality in the endpoint.

Thanks! Looks great

**Pull request merged**